apps/website/src/components/courses/ResourceListItem.tsx

```diff
@@ -121,46 +120,34 @@ export const ResourceListItem: React.FC<ResourceListItemProps> = ({ resource })
   const [hasCompletionLoaded, setHasCompletionLoaded] = useState(false);
   const [isHovered, setIsHovered] = useState(false);
 
-  // Fetch resource completion data
-  const [{
+  // Fetch resource completion data (only when authenticated)
+  const {
     data: completionData,
-    loading: completionLoading,
+    isLoading: completionLoading,
     error: completionError,
-  }, fetchResourceCompletion] = useAxios<GetResourceCompletionResponse>({
-    method: 'get',
-    url: `/api/courses/resource-completion/${resource.id}`,
-    headers: {
-      Authorization: `Bearer ${auth?.token}`,
+    refetch: fetchResourceCompletion,
+  } = trpc.resources.getResourceCompletion.useQuery(
+    { unitResourceId: resource.id },
+    {
+      enabled: !!auth,
+      retry: false, // Don't retry on 404
```

Comment:
This could potentially mask network failures etc that should be retried. You can use `refetchInterval` to check the error code and conditionally not retry on 404 (see https://github.com/bluedotimpact/bluedot/pull/1463/files#diff-005ce46ba7f01afbf111842b7b35ea341230e496a4450e23d178e329fd8cef8bR35).